src/snowflake/snowpark/_internal/utils.py

```diff
@@ -423,14 +444,19 @@ def calculate_checksum(
                 # ignore generated python files
                 if ignore_generated_py_file and file.endswith(GENERATED_PY_FILE_EXT):
                     continue
+
                 hash_algo.update(file.encode("utf8"))
+
+                filename = os.path.join(dirname, file)
+                file_size = os.path.getsize(filename)
+
+                if whole_file_hash:
+                    _hash_file(hash_algo, filename, chunk_size, whole_file_hash)
+                    current_size += file_size
                 if current_size < chunk_size:
-                    filename = os.path.join(dirname, file)
-                    file_size = os.path.getsize(filename)
                     read_size = min(file_size, chunk_size - current_size)
                     current_size += read_size
-                    with open(filename, "rb") as f:
-                        hash_algo.update(f.read(read_size))
+                    _hash_file(hash_algo, filename, read_size, False)
```

Comment:
just checking if we have a test case that covers this part of the code. i.e. chunk_size > file_size

Reply:
Good call. Most of the existing examples in the test for this code had smaller file size than chunk size.